src/main/java/nl/b3p/tailormap/api/repository/events/OIDCConfigurationEventHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2023 B3Partners B.V.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+package nl.b3p.tailormap.api.repository.events;
+
+import nl.b3p.tailormap.api.persistence.OIDCConfiguration;
+import nl.b3p.tailormap.api.security.OIDCRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.core.annotation.HandleAfterCreate;
+import org.springframework.data.rest.core.annotation.HandleAfterDelete;
+import org.springframework.data.rest.core.annotation.HandleAfterSave;
+import org.springframework.data.rest.core.annotation.HandleBeforeCreate;
+import org.springframework.data.rest.core.annotation.HandleBeforeSave;
+import org.springframework.data.rest.core.annotation.RepositoryEventHandler;
+import org.springframework.stereotype.Component;
+
+@RepositoryEventHandler
+@Component
+public class OIDCConfigurationEventHandler {
+  @Autowired OIDCRepository oidcRepository;
+
+  public OIDCConfigurationEventHandler() {}
+
+  @HandleBeforeCreate
+  @HandleBeforeSave
+  public void handleBeforeCreateOrSave(OIDCConfiguration configuration) throws Exception {
+    // If the user provided a "full" OIDC discovery URL, strip it.
+    if (configuration.getIssuerUrl().endsWith("/.well-known/openid-configuration")) {
+      String issuerUrl = configuration.getIssuerUrl();
+      issuerUrl =
+          issuerUrl.substring(0, issuerUrl.lastIndexOf("/.well-known/openid-configuration"));
+      configuration.setIssuerUrl(issuerUrl);
+    }
+  }
+
+  @HandleAfterCreate
+  @HandleAfterSave
+  @HandleAfterDelete
+  public void handleAfterCreateOrSaveOrDelete(OIDCConfiguration configuration) throws Exception {
+    oidcRepository.synchronize();
```

Comment:
This refreshes all configurations instead of only the changed configuration, which in a way is nice because you want a way to refresh a configuration other than restarting, but it is not explicit that you refresh all configurations by create/updating/deleting any config.

Reply:
OK. My experience with SAML tells me SSO configurations can be very brittle, but let's hope OIDC isn't... I guess OIDC as implemented by default is more like SAML without automatic metadata import from URL (which some implementations do but we didn't do).